Enable GetDimensions test for Clang & DirectX

hekota opened the pull request:
These tests are now passing for DirectX. The SPIR-V backend does not yet support the intrinsic needed for this test, so the link now points to a new issue that is SPIR-V only.

spall reviewed (approved):
I guess I'd just suggest updating the pr description to specify they're not passing on clang vulkan

hekota commented:
My apologies, you have every right to be confused! I made a terrible typo in the PR description ("not passing" vs. "now passing"), and I did not explain the Vulkan part. In my mind this was "remove an XFAIL" change since the tests are now passing on Clang and DirectX. But then I realized Vulkan does not have support for the needed intrinsic yet, so I added the XFAIL back with a new issue link for Vulkan only. The description is now updated.

damyanp reviewed (approved):
LGTM, thanks!
